process/smartContract/processorV2/vmOutputAccountsProcessor.go

```diff
@@ -0,0 +1,314 @@
+package processorV2
+
+import (
+	"bytes"
+	"math/big"
+
+	"github.com/ElrondNetwork/elrond-go-core/core"
+	"github.com/ElrondNetwork/elrond-go-core/core/check"
+	"github.com/ElrondNetwork/elrond-go-core/data"
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/ElrondNetwork/elrond-go/state"
+	vmcommon "github.com/ElrondNetwork/elrond-vm-common"
+)
+
+// VMOutputAccountsProcessor will process the VMOutput from a regular of builtin function call
+type VMOutputAccountsProcessor struct {
+	sc       *scProcessor
+	vmInput  *vmcommon.VMInput
+	vmOutput *vmcommon.VMOutput
+	tx       data.TransactionHandler
+	txHash   []byte
+
+	processOutputTransfers bool
+	processStorageUpdates  bool
+	processCode            bool
+	processBalanceDelta    bool
+	processNonce           bool
+}
+
+// NewVMOutputAccountsProcessor creates a new VMOutputProcessor instance.
+func NewVMOutputAccountsProcessor(
+	sc *scProcessor,
+	vmInput *vmcommon.VMInput,
+	vmOutput *vmcommon.VMOutput,
+	tx data.TransactionHandler,
+	txHash []byte) *VMOutputAccountsProcessor {
+	return &VMOutputAccountsProcessor{
+		sc:       sc,
+		vmInput:  vmInput,
+		vmOutput: vmOutput,
+		tx:       tx,
+		txHash:   txHash,
+
+		processOutputTransfers: false,
+		processStorageUpdates:  false,
+		processCode:            false,
+		processBalanceDelta:    false,
+		processNonce:           false,
+	}
+}
+
+// Run runs the configured processing steps on vm output
+func (oap *VMOutputAccountsProcessor) Run() (bool, []data.TransactionHandler, error) {
```

Comment:
This is way too much to review if it is ok or not. You merged the functionality for the other part of the sc processor into this function as to have an all-in-one solution to all cases based on some flags. We need to find a better solution to this.

Reply:
fixed, we now have only one variant of processor that handles both post built in and non-built in calls